src/battle-dex.ts

```diff
@@ -601,7 +601,7 @@ const Dex = new class implements ModdedDex {
 
 		if (animationData[facing + 'f'] && options.gender === 'F') facing += 'f';
 		let allowAnim = !Dex.prefs('noanim') && !Dex.prefs('nogif');
-		if (allowAnim && genNum >= 6) spriteData.pixelated = false;
+		if (fieldGenNum === 5 || allowAnim && genNum >= 6) spriteData.pixelated = false;
```

Comment:
Can you help me understand `fieldGenNum` vs `genNum`? Its not clear to me why we check one vs the other in this function (like below - most code in the `options.noScale` block checks `fieldGenNum` but not L640). Understanding the distinction here (and how it relates to the `BattleScene.gen`) is important for https://github.com/Zarel/Pokemon-Showdown-Client/issues/1369 as well

Reply:
I'm surprised it wasn't clear from the definition.

`genNum` is the actual generation of the battle, for the purposes of game mechanics.

`fieldGenNum` is the generation we pull the sprites for the battle from (it also affects what backdrop we're using).

Reply:
`options.gen` is the battle gen AFAICT?

```
let genNum = Math.max(fieldGenNum, Math.min(template.gen, 5));
```

is gen num, and I'm not entirely sure what it does. Also, why would the battle gen effect y position etc?

Reply:
Also, I feel `spriteGenNum` is perhaps more clear than `fieldGenNum`?

Reply:
Oh, I didn't look at `genNum`. That's the sprite generation for the individual sprite. It's capped so we don't try to grab e.g. a gen-2 sprite for a gen-6 pokemon.

Reply:
So yeah `spriteGenNum` would be a better name for clarity there.

Reply:
Are we going to do the rename?

Reply:
I'll rename this during #1369